include/dlaf/eigensolver/tridiag_solver/merge.h

```diff
@@ -458,28 +491,30 @@ inline pika::future<SizeType> stablePartitionIndexForDeflation(SizeType i_begin,
                                                                          std::move(sender)));
 }
 
-inline void setColTypeTile(const matrix::Tile<ColType, Device::CPU>& tile, ColType val) {
-  SizeType len = tile.size().rows();
-  for (SizeType i = 0; i < len; ++i) {
-    tile(TileElementIndex(i, 0)) = val;
-  }
-}
-
-DLAF_MAKE_CALLABLE_OBJECT(setColTypeTile);
-DLAF_MAKE_SENDER_ALGORITHM_OVERLOADS(dlaf::internal::TransformDispatchType::Plain, setColTypeTile,
-                                     setColTypeTile_o)
-
+template <Device D>
 inline void initColTypes(SizeType i_begin, SizeType i_split, SizeType i_end,
-                         Matrix<ColType, Device::CPU>& coltypes) {
-  using dlaf::internal::Policy;
-  using dlaf::internal::whenAllLift;
-  using pika::execution::thread_priority;
-  using pika::execution::experimental::start_detached;
+                         Matrix<ColType, D>& coltypes) {
+  namespace ex = pika::execution::experimental;
+  namespace di = dlaf::internal;
 
   for (SizeType i = i_begin; i <= i_end; ++i) {
     ColType val = (i <= i_split) ? ColType::UpperHalf : ColType::LowerHalf;
-    whenAllLift(coltypes.readwrite_sender(LocalTileIndex(i, 0)), val) |
-        setColTypeTile(Policy<Backend::MC>(thread_priority::normal)) | start_detached();
+
+    auto set_fn = [val](const auto& ct_tile, [[maybe_unused]] auto&&... ts) {
+      if constexpr (D == Device::CPU) {
+        for (SizeType i = 0; i < ct_tile.size().rows(); ++i) {
+          ct_tile(TileElementIndex(i, 0)) = val;
+        }
+      }
+      else {
+        setColTypeTile(val, ct_tile.size().rows(), ct_tile.ptr(), ts...);
+      }
+    };
+
+    ex::start_detached(
+        di::transform<di::TransformDispatchType::Plain>(di::Policy<DefaultBackend<D>::value>(),
```

Comment:
Side note, `TransformDispatchType::Plain` is the default so if you want you can leave it out here and in many other places (except if you need to disable unwrapping).

Reply:
https://github.com/eth-cscs/DLA-Future/pull/617/commits/3a0c23bdae5064dd60f15e52074825dd14ef270b